--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-common.dtsi
@@ -29,6 +29,24 @@
 
 &firmware {
 	android {
-		/delete-node/ vbmeta;
+		fstab {
+			compatible = "android,fstab";
+			system {
+				compatible = "android,system";
+				dev = "/dev/block/platform/soc/1d84000.ufshc/by-name/system";
+				type = "ext4";
+				mnt_flags = "ro,barrier=1,discard";
+				fsmgr_flags = "wait";
+				status = "ok";
+			};
+			vendor {
+				compatible = "android,vendor";
+				dev = "/dev/block/platform/soc/1d84000.ufshc/by-name/vendor";
+				type = "ext4";
+				mnt_flags = "ro,barrier=1,discard";
+				fsmgr_flags = "wait";
+				status = "ok";
+			};
+		};
 	};
 };
