--- a/arch/arm64/boot/dts/qcom/sdm845-v2.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-v2.dtsi
@@ -573,7 +573,7 @@
 
 		qcom,gpu-pwrlevel@0 {
 			reg = <0>;
-			qcom,gpu-freq = <710000000>;
+			qcom,gpu-freq = <820000000>;
 			qcom,bus-freq = <12>;
 			qcom,bus-min = <12>;
 			qcom,bus-max = <12>;
@@ -581,25 +581,25 @@
 
 		qcom,gpu-pwrlevel@1 {
 			reg = <1>;
-			qcom,gpu-freq = <675000000>;
+			qcom,gpu-freq = <710000000>;
 			qcom,bus-freq = <12>;
-			qcom,bus-min = <10>;
+			qcom,bus-min = <11>;
 			qcom,bus-max = <12>;
 		};
 
 		qcom,gpu-pwrlevel@2 {
 			reg = <2>;
 			qcom,gpu-freq = <596000000>;
-			qcom,bus-freq = <10>;
-			qcom,bus-min = <9>;
+			qcom,bus-freq = <11>;
+			qcom,bus-min = <10>;
 			qcom,bus-max = <12>;
 		};
 
 		qcom,gpu-pwrlevel@3 {
 			reg = <3>;
 			qcom,gpu-freq = <520000000>;
-			qcom,bus-freq = <9>;
-			qcom,bus-min = <8>;
+			qcom,bus-freq = <10>;
+			qcom,bus-min = <9>;
 			qcom,bus-max = <11>;
 		};

--- a/drivers/clk/qcom/gpucc-sdm845.c
+++ b/drivers/clk/qcom/gpucc-sdm845.c
@@ -280,8 +280,8 @@ static const struct freq_tbl  ftbl_gpu_cc_gx_gfx3d_clk_src_sdm845_v2[] = {
 	F(414000000, P_CRC_DIV,  1, 0, 0),
 	F(520000000, P_CRC_DIV,  1, 0, 0),
 	F(596000000, P_CRC_DIV,  1, 0, 0),
-	F(675000000, P_CRC_DIV,  1, 0, 0),
 	F(710000000, P_CRC_DIV,  1, 0, 0),
+	F(820000000, P_CRC_DIV,  1, 0, 0),
 	{ }
 };
 
@@ -637,9 +637,9 @@ static void gpu_cc_gfx_sdm845_fixup_sdm845v2(void)
 		520000000;
 	gpu_cc_gx_gfx3d_clk_src.clkr.hw.init->rate_max[VDD_GX_NOMINAL_L1] =
 		596000000;
-	gpu_cc_gx_gfx3d_clk_src.clkr.hw.init->rate_max[VDD_GX_HIGH] = 675000000;
+	gpu_cc_gx_gfx3d_clk_src.clkr.hw.init->rate_max[VDD_GX_HIGH] = 710000000;
 	gpu_cc_gx_gfx3d_clk_src.clkr.hw.init->rate_max[VDD_GX_HIGH_L1] =
-		710000000;
+		820000000;
 }
 
 static void gpu_cc_gfx_sdm845_fixup_sdm670(void)
